UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -208,5 +209,43 @@ public string GenerateQueryForFss(List<Products> products)
             sb.Append(")");//// last main )
             return sb.ToString();
         }
+
+        public async Task<bool> DownloadBatchFiles(IEnumerable<string> uri, string downloadPath)
+        {
+            HttpResponseMessage httpResponse;
+            string payloadJson = string.Empty;
+            bool result = false;
+            var accessToken = await authTokenProvider.GetManagedIdentityAuthAsync(fileShareServiceConfig.Value.ResourceId);
+
+            foreach (var item in uri)
+            {
+                httpResponse = await fileShareServiceClient.CallFileShareServiceApi(HttpMethod.Get, payloadJson, accessToken, item);
+                var fileName = item.Split("/").Last();
+                if (httpResponse.IsSuccessStatusCode)
+                {
+                    if (!Directory.Exists(downloadPath))
+                    {
+                        Directory.CreateDirectory(downloadPath);
+                    }
+                    string path = Path.Combine(downloadPath, fileName);
+                    if (!File.Exists(path))
+                    {
+                        using (Stream stream = await httpResponse.Content.ReadAsStreamAsync())
+                        {
+                            using (FileStream outputFileStream = new FileStream(path, FileMode.Create, FileAccess.ReadWrite))
+                            {
+                                stream.CopyTo(outputFileStream);
+                                result = true;
+                            }
+                        } 
+                    }
+                }
+                else
+                {
+                    logger.LogInformation(EventIds.DownloadFileShareServiceNonOkResponse.ToEventId(), "File share service download end point with uri {RequestUri} responded with {StatusCode}", httpResponse.RequestMessage.RequestUri, httpResponse.StatusCode);
+                }
```

Comment:
This code could be broken into a number of methods which self document its purpose and would make it easier for a support engineer to understand.

For example the outer foreach loop could contain a single method called for example  "ProcessBatchFile". The second if statement could be a method "CheckCreateFolderPath" and the third piece of work a method called "CopyFileToFolder".  I'm not entirely sure these would be the best names but hopefully the names would reflect the work that the code is doing.

Reply:
Yes split methods respectively for clear understanding